Automattic-Tracks-iOS/Services/EventLoggingNetworkService.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+typealias NetworkResultCallback = (Result<Data?, Error>) -> Void
+
+open class EventLoggingNetworkService {
```

Comment:
We could use a `protocol`, but when it's time to use this, we'd need a concrete implementation. It'd be possible to confirm an object to this protocol and thus adopt this method as a default implementation, but I'm not sure I'm a fan of that.

The `open` part just allows other packages (like the test package) to subclass this – classes in Swift are `final` by default outside their package, which is why this one is open.

Reply:
> The open part just allows other packages (like the test package) to subclass this – classes in Swift are final by default outside their package, which is why this one is open.

Yeap, I know about that part and that is what was bothering me. I don't like implementation inheritance at all and strongly believe there is always a better way to do it. Having said that, I don't try and enforce this subjective opinion and just bring it up and leave it to the developer. If you want to talk about this further I am happy to, but otherwise if you're happy we can totally leave it as is.

Reply:
I went over this a bunch and tried with a protocol / implementation pair. While it works, it felt icky at this point since these classes are still `internal` and can't be overridden.

What I've done to simplify _this_ PR is to remove the `open` modifier on these classes. Because they can be imported into `XCTest` using `@testable`, which treats it as though it's `internal` to the test module,  it's effectively `internal open class`, which isn't useful to anyone.

Removing the `open` modifier has no effect on the tests, and if we need to make it `public` in a subsequent PR, I think it'll make sense to expose a `public` protocol with an `internal` default implementation at that point.

Reply:
Oh – changed in a8fb2c03bf581e846288fa229bccfc26195265ed.